pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarFetcher.java

```diff
@@ -84,10 +86,9 @@
 
     /**
      * Optional watermark strategy that will be run per pulsar partition, to exploit per-partition
-     * timestamp characteristics. The watermark strategy is kept in serialized form, to deserialize
-     * it into multiple copies.
+     * timestamp characteristics.
      */
-    private final SerializedValue<WatermarkStrategy<T>> watermarkStrategy;
+    private final WatermarkStrategy<T> watermarkStrategy;
```

Comment:
The watermark strategy may be deserialized at this stage because the `PulsarFetcher` is instantiated on the task manager.